packages/docusaurus-theme-classic/src/theme/hooks/useHideableNavbar.ts

```diff
@@ -5,22 +5,25 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import {useState, useCallback, useEffect, useRef} from 'react';
+import {useState, useEffect, useRef} from 'react';
 import {useLocation} from '@docusaurus/router';
+import {useThemeConfig, domUtils} from '@docusaurus/theme-common';
 import useScrollPosition from '@theme/hooks/useScrollPosition';
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
 import type {useHideableNavbarReturns} from '@theme/hooks/useHideableNavbar';
 
-const useHideableNavbar = (hideOnScroll: boolean): useHideableNavbarReturns => {
+const useHideableNavbar = (): useHideableNavbarReturns => {
+  const {hideOnScroll} = useThemeConfig().navbar;
   const location = useLocation();
-  const [isNavbarVisible, setIsNavbarVisible] = useState(hideOnScroll);
   const isFocusedAnchor = useRef(false);
+  const [navbarVisible, setNavbarVisible] = useState(hideOnScroll);
   const [lastScrollTop, setLastScrollTop] = useState(0);
-  const [navbarHeight, setNavbarHeight] = useState(0);
-  const navbarRef = useCallback((node: HTMLElement | null) => {
-    if (node !== null) {
-      setNavbarHeight(node.getBoundingClientRect().height);
-    }
-  }, []);
+  const navbarHeight =
+    (ExecutionEnvironment.canUseDOM &&
+      domUtils.convertRemToPx(
+        domUtils.getPropertyValue('--ifm-navbar-height'),
+      )) ||
+    0;
```

Comment:
Does it give a different height compared to before?
I feel this solution is more complex, more likely to break (for example if we change the rem unit to something else) and more expensive to run.

Isn't reading the stylesheet in render expensive?

Reply:
As understand, these functions have about the same effect on performance, but I agree with you, so I simplified the solution as you suggested above.